lib/sftp.js

```diff
@@ -2869,10 +2869,10 @@ ReadStream.prototype._read = function(n) {
     }
 
     this.push(b);
-  });
 
-  // Move the pool positions, and internal position for reading.
-  this.pos += toRead;
+    // Move the pool positions, and internal position for reading.
+    this.pos += b && b.length || 0;
```

Comment:
Does `this.bytesRead` need to be adjusted also? Or is bytesRead always accurate? If so then maybe use `bytesRead` instead of `b && b.length`.

Reply:
It looks like `bytesRead` is accurate. I'll update. :+1: